jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -236,36 +236,33 @@ private void saveDocker() {
     Preconditions.checkArgument(dockerClient.isPresent());
     ProgressEventDispatcher.Factory childProgressDispatcherFactory =
         Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
-    assignLocalImageResult(
+    Future<LocalImage> localImage =
         executorService.submit(
-            LocalBaseImageSteps.retrieveDockerDaemonImageStep(
-                buildConfiguration, childProgressDispatcherFactory, dockerClient.get())));
+            LocalBaseImageSteps.retrieveDockerDaemonLayersStep(
+                buildConfiguration,
+                childProgressDispatcherFactory,
+                dockerClient.get(),
+                tempDirectoryProvider));
+    results.baseImageLayers = executorService.submit(() -> localImage.get().layers);
+    results.baseImageAndAuth =
+        executorService.submit(LocalBaseImageSteps.retrieveImageAndAuthorizationStep(localImage));
```

Comment:
I expect this to be at least

```java
executorService.submit(
    () -> LocalBaseImageSteps.retrieveImageAndAuthorizationStep(localImage.get()));
```

to avoid putting `Future` in the method signature of `retrieveImageAndAuthorizationStep()`. But ultimately,

```java
executorService.submit(
    () -> {
      List<PreparedLayer> layers = realizeFutures(localImage.get().layers);
      return LocalBaseImageSteps.retrieveImageAndAuthorizationStep(layers, localImage.get().containerConfiguration);
    }
```

and `retrieveImageAndAuthorizationStep()` shouldn't block or deal with anything related to concurrency like `Future` or `ExecutorService`.